signbank/dictionary/update.py

```diff
@@ -764,6 +771,345 @@ def confirm_import_gloss_csv(request):
+                                Gloss.semantic_field.through(
+                                    gloss_id=gloss.id,
+                                    fieldchoice_id=semantic_fields_dict["Miscellaneous"]
+                                )
+                            )
+
+                    # Prepare new idgloss fields for bulk update
+                    gloss.idgloss = f"{old_id_gloss[0]}:{gloss.pk}"
+                    if gloss.idgloss_mi:
+                        gloss.idgloss_mi = f"{gloss.idgloss_mi}:{gloss.pk}"
+                    bulk_update_glosses.append(gloss)
+
+                    translations.append(GlossTranslations(
+                        gloss=gloss,
+                        language=language_en,
+                        translations=gloss_data["word"],
+                        translations_secondary=gloss_data.get("secondary", None)
+                    ))
+                    if gloss_data.get("maori", None):
+                        translations.append(GlossTranslations(
+                            gloss=gloss,
+                            language=language_mi,
+                            translations=gloss_data["maori"]
+                        ))
+
+                    comments.append(Comment(
+                        content_type=gloss_content_type,
+                        object_pk=gloss.pk,
+                        user_name=gloss_data.get("contributor_name", ""),
+                        user_email=gloss_data.get("contributor_email", ""),
+                        comment=gloss_data.get("notes", ""),
+                        site=site,
+                        is_public=False,
+                        submit_date=comment_submit_date
+                    ))
+                    if gloss_data.get("sign_comments", None):
+                        for comment in gloss_data["sign_comments"].split("|"):
+                            try:
+                                comment_content = comment.split(":")
+                                user_name = comment_content[0]
+                                comment_content = comment_content[1]
+                            except IndexError:
+                                comment_content = comment
+                                user_name = "Unknown"
+                            comments.append(Comment(
+                                content_type=gloss_content_type,
+                                object_pk=gloss.pk,
+                                user_name=user_name,
+                                comment=comment_content,
+                                site=site,
+                                is_public=False,
+                                submit_date=comment_submit_date
+                            ))
+
+                    if gloss_data.get("videos", None):
+                        for i, video_url in enumerate(gloss_data["videos"].split("|")):
+                            extension = video_url[-3:]
+                            file, _ = urlretrieve(
```

Comment:
**issue**: retrieving all the assets will likely take a significant time. If possible, this should be moved to a background task instead of occurring in the request.